state-sync/src/coordinator.rs

```diff
@@ -180,40 +180,44 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
         client_events: mpsc::UnboundedReceiver<CoordinatorMessage>,
         state_sync_to_mempool_sender: mpsc::Sender<diem_mempool::CommitNotification>,
         network_senders: HashMap<NodeNetworkId, StateSyncSender>,
-        role: RoleType,
+        node_config: &NodeConfig,
         waypoint: Waypoint,
-        config: StateSyncConfig,
-        upstream_config: UpstreamConfig,
         executor_proxy: T,
         initial_state: SyncState,
     ) -> Result<Self> {
         info!(LogSchema::event_log(LogEntry::Waypoint, LogEvent::Initialize).waypoint(waypoint));
+
+        // Create a new request manager.
+        let role = node_config.base.role;
+        let tick_interval_ms = node_config.state_sync.tick_interval_ms;
         let retry_timeout_val = match role {
-            RoleType::FullNode => config
-                .tick_interval_ms
-                .checked_add(config.long_poll_timeout_ms)
+            RoleType::FullNode => tick_interval_ms
+                .checked_add(node_config.state_sync.long_poll_timeout_ms)
                 .ok_or_else(|| format_err!("Fullnode retry timeout has overflown."))?,
-            RoleType::Validator => config
-                .tick_interval_ms
+            RoleType::Validator => tick_interval_ms
                 .checked_mul(2)
                 .ok_or_else(|| format_err!("Validator retry timeout has overflown!"))?,
```

Comment:
@zekun000 is there a reason we cannot unify this?

Reply:
Thanks @davidiw! Will wait to see what @zekun000 says about this!

Reply:
is there any issue just using what FNs use?

Reply:
ahh, maybe we just leave well enough alone here, since I imagine an adaptive approach is probably more intelligent... probably not worth the headache here.

Reply:
because FN is using long poll subscription but validator is doing RPC so different timeout value makes sense (long poll timeout is 10s)

Reply:
Ack, I'll leave this as it is for now and we can think about something cleaner if it makes sense in the future!